src/compat/eme/generate_key_request.ts

```diff
@@ -79,8 +94,35 @@ export function patchInitData(initData : Uint8Array) : Uint8Array {
         initData[offset + 26] === 0xFB &&
         initData[offset + 27] === 0x4B
     ) {
-      log.info("Compat: CENC PSSH found.");
-      cencs = concat(cencs, currentPSSH);
+      const cencOffsets = getNextBoxOffsets(currentPSSH);
+      const version = cencOffsets === null ? undefined :
+                                             currentPSSH[cencOffsets[1]];
+      log.info("Compat: CENC PSSH found with version", version);
+      if (version === undefined) {
```

Comment:
Ever considered using an if-else-if construct (suggestion below) ? I know some people find it less readable. It is probably a matter of taste (and C-coding habits) ...
```suggestion
if (version === undefined) {
    log.warn("Compat: could not read version of CENC PSSH");
} else if (encounteredCencs.length === 0) {
    encounteredCencs.push({ version, pssh: currentPSSH });
} else if (hasV1 = encounteredCencs.some(cenc => cenc.version === 1),
           isV1 = version === 1,
           hasV1 == isV1 ) {
    encounteredCencs.push({ version, pssh: currentPSSH });
} else if ( isV1 ) {
    log.warn("Compat: cenc version 1 encountered, removing every other cenc pssh box.");
    encounteredCencs = [{ version, pssh: currentPSSH }];
} else {
    log.warn("Compat: filtering out cenc pssh box with wrong version", version);
}

```

Reply:
I realize now that I've written a lot of imbricated ifs here!
It's true that it might be more readable by transforming it the way you wrote it.

But there is some things in your example that are not idiomatic to a JS developer, most notably setting `hasV1` and `isV1` in an `else if` clause and reusing `isV1` as another branch's condition doesn't seem "natural" to a JS/TS developer and would look a little out of place when comparing with the rest of the code.

Here, the `isV1` variable will also have to be defined (through a `let isV1` statement) in a scope encompassing both of its uses (so on top of the first `if`) which might also be unusual (and might be difficult to properly type, as I'm not sure that TypeScript will be able to know that it is always initialized to a boolean when we check it in the other `else if`).

I'll still try to flatten a little that code (for example, the first `else if` looks better to me than what I wrote) but while keeping some else's aside for creating a new scope in which related variables can be initialized (again, to stay idiomatic with the JS/TS way of doing things).

Reply:
OK, thanks for the explanation. I understand this is not idiomatic in JS/TS coding. Actually, most C coders don't like these comma-separated statements inside `if` clauses either. FWIW, I find it useful (in C) when writing `if-else-if` hierarchized conditionnal processings to defer intermediate computations to the last possible step (i.e. only after they become computable and not before they are actually required) as it allows some optimizations while remaining decently readable (at least in terms nesting/depth, once you get used to it).

Reply:
Yes it does seem more readable.

I decided to remove the need of that debate anyway by just creating a `foundCencV1` boolean instead of always iterating to see if we've detected a v1 already (which seems more readable and more efficient anyway).

I used your "trick" for merging two branches by checking `foundCencV1 === (version === 1)`. It does seem a little too smart so I still hesitated (we usually optimize for readability in the RxPlayer), but  with a comment and the nesting it save it actually seems more readable.

Thanks for sharing that code by the way. Even in the case where we could not apply it, I find it is always nice to have some opinions about how the same code could be written in another language / with another mindset. We're a small team so we have few outside influence on our code style, so I consider comments like that a gain for our code.
And I also appreciate those personally as - even if we're speaking about a pretty generic code logic here - I do have some interests about lower-level languages.

Reply:
Yes, I believe that code compactness is generally a quality (ability to consider a whole logic block at a glance, do things only once / "single point of truth", fewer lines=fewer bugs, etc.), as long as the logic of the code remains obvious to the reader. Anyway, your latest code change seems perfect to me.